### ISSUE TITLE
Adding support for difference between exists and empty

### ISSUE THREAD

**Issue by Tilogorn** [eivindfjeldstad/validate#32]
Take the following MWE:

```
var schema = require('validate');

var my_scheme = schema({
    name: {
        required: true,
        type: 'string'
    }
});

var case1 = my_scheme.validate({name: 'John'});
var case2 = my_scheme.validate({name: ''});

console.log(case1); // returns empty array
console.log(case2); // returns error
```

The property is set (that is what I understand when setting `required: true`) but the value is empty and the validation fails. For me thats an unexpected behaviour.

Is there a way to check just for "property exists"?

**Comment by eivindfjeldstad** [comment]
The way required works has changed somewhat in 4.0, mainly for booleans and numbers, but an empty string is still considered invalid. It seems to me that most people are passing in objects constructed from request bodies etc., so having an empty string not pass validation makes sense in that case. I'm open to changing this in a future release though.

**Comment by eivindfjeldstad** [comment]
Yeah, I agree it's not ideal. It's also a road block for any sane default value implementation (#33). I like the idea of separating `required` from `empty`.
